packages/cli/src/agent.test.ts

```diff
@@ -0,0 +1,1326 @@
+import {
+  allOk,
+  badRequest,
+  ContentType,
+  indexSearchParameterBundle,
+  indexStructureDefinitionBundle,
+} from '@medplum/core';
+import { readJson, SEARCH_PARAMETER_BUNDLE_FILES } from '@medplum/definitions';
+import { Agent, Bundle, Parameters, Reference, SearchParameter } from '@medplum/fhirtypes';
+import { MockClient } from '@medplum/mock';
+import { randomUUID } from 'node:crypto';
+import { main } from '.';
+import { createMedplumClient } from './util/client';
+
+const EXAMPLE_HL7_MSG = `MSH|^~\\&|SENDING_APPLICATION|SENDING_FACILITY|RECEIVING_APPLICATION|RECEIVING_FACILITY|20240927120000||ADT^A01|MSG00001|P|2.3
+EVN|A01|20240927120000
+PID|1||123456789^^^HOSPITAL^MR||Doe^John^^^^||19800101|M`;
+
+jest.mock('./util/client');
+jest.mock('node:fs', () => ({
+  existsSync: jest.fn(),
+  readFileSync: jest.fn(),
+  writeFileSync: jest.fn(),
+  constants: {
+    O_CREAT: 0,
+  },
+  promises: {
+    readFile: jest.fn(async () => '{}'),
+  },
+}));
+
+describe('Agent CLI', () => {
+  const env = process.env;
+  let processError: jest.SpyInstance;
+  let consoleTableSpy: jest.SpyInstance;
+  let consoleInfoSpy: jest.SpyInstance;
+  let consoleErrorSpy: jest.SpyInstance;
+  let medplum: MockClient;
+  let medplumGetSpy: jest.SpyInstance;
+
+  beforeAll(() => {
+    process.exit = jest.fn<never, any>().mockImplementation(function exit(exitCode: number) {
+      throw new Error(`Process exited with exit code ${exitCode}`);
+    }) as unknown as typeof process.exit;
+    processError = jest.spyOn(process.stderr, 'write').mockImplementation(jest.fn());
+    consoleTableSpy = jest.spyOn(console, 'table').mockImplementation(jest.fn());
+    consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(jest.fn());
+    consoleInfoSpy = jest.spyOn(console, 'info').mockImplementation(jest.fn());
+
+    indexSearchParameterBundle(readJson('fhir/r4/profiles-types.json') as Bundle<SearchParameter>);
+    indexStructureDefinitionBundle(readJson('fhir/r4/profiles-resources.json') as Bundle);
+    indexStructureDefinitionBundle(readJson('fhir/r4/profiles-medplum.json') as Bundle);
+    for (const filename of SEARCH_PARAMETER_BUNDLE_FILES) {
+      indexSearchParameterBundle(readJson(filename) as Bundle<SearchParameter>);
+    }
+  });
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+    medplum = new MockClient();
+    (createMedplumClient as unknown as jest.Mock).mockImplementation(async () => medplum);
+
+    process.env = { ...env };
+    medplumGetSpy = jest.spyOn(medplum, 'get');
+  });
+
+  afterEach(() => {
+    process.env = env;
```

Comment:
Are both these assignments to `process.env` needed?

Reply:
I think this `afterEach` should actually be an `afterAll` probably, to reset the `process.env` at the end